collector/ntp.go

```diff
@@ -17,62 +17,145 @@ package collector
 
 import (
 	"fmt"
+	"net"
+	"time"
 
 	"github.com/beevik/ntp"
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/common/log"
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
+const (
+	hour24 = 24 * time.Hour // `time` does not export `Day` as Day != 24h because of DST
+)
+
 var (
-	ntpServer          = kingpin.Flag("collector.ntp.server", "NTP server to use for ntp collector.").Default("").String()
+	ntpServer          = kingpin.Flag("collector.ntp.server", "NTP server to use for ntp collector").Default("127.0.0.1").String()
 	ntpProtocolVersion = kingpin.Flag("collector.ntp.protocol-version", "NTP protocol version").Default("4").Int()
+	ntpServerIsLocal   = kingpin.Flag("collector.ntp.server-is-local", "Certify that collector.ntp.server address is the same local host as this collector.").Default("false").Bool()
```

Comment:
Should this be `true` by default?

Reply:
Nope, default IP passes `IsLoopback` check. That's safety check to disarm if you want to poll something like 172.16.0.1.